nixos/modules/services/misc/nix-daemon.nix

```diff
@@ -620,8 +620,17 @@ in
 
     environment.extraInit =
       ''
-        if [ -e "$HOME/.nix-defexpr/channels" ]; then
-          export NIX_PATH="$HOME/.nix-defexpr/channels''${NIX_PATH:+:$NIX_PATH}"
+        NIX_DEFEXPR="$HOME/.nix-defexpr"
+        if ! [ -e "$NIX_DEFEXPR" ]; then
+            if [ -n "$XDG_DATA_HOME" ]; then
+                NIX_DEFEXPR="$XDG_DATA_HOME/nix/defexpr"
+            else
+                NIX_DEFEXPR="$HOME/.local/share/nix/defexpr"
+            fi
+        fi
```

Comment:
logic like this could be useful for setting the nix-profile so users could just use that env var in their scripts instead of having to accommodate 3 possible separate locations

`NIX_PROFILE_PATH` perhaps?

Reply:
Nix has a setup script that has to be sourced in order to bring nix into the PATH, with the proper ssl certs for packages to use. So that would be a good place to define the variable.

Reply:
That would require an update to be simultaneous in both Nix and nixpkgs, which is a lot more difficult to orchestrate correctly.